packages/react-router/index.tsx

```diff
@@ -543,6 +541,13 @@ function useRoutes_(
     ? joinPaths([parentPathname, basename])
     : parentPathname;
 
+  let contextLocation = useLocation();
+  let location = React.useMemo(() => {
+    return locationOverride
+      ? { ...contextLocation, ...locationOverride }
```

Comment:
I guess the purpose of doing it this way is so you can specify just a query string in your location override but still use the pathname coming through on context, right? I'm thinking something like

```tsx
<Routes location={{ search: '?brand=adidas' }}>
```

I wonder what the use case would be there. Seems like most of the time you're overriding the location you're being very explicit about it, no? Do you want to still get extra stuff from context?

Reply:
Just to recap our in-person convo: let's default the `pathname` to `/` but still tell them in the type declaration that the `pathname` should be a string, so hopefully they give us one.